cpp/src/arrow/compute/kernels/hash.cc

```diff
@@ -415,5 +506,31 @@ Status DictionaryEncode(FunctionContext* ctx, const Datum& value, Datum* out) {
   return Status::OK();
 }
 
+const char kValuesFieldName[] = "Values";
+const char kCountsFieldName[] = "Counts";
```

Comment:
I have a personal preference for lowercase, but I don't think it's too important for now